indico/htdocs/js/indico/jquery/colorswitch.js

```diff
@@ -0,0 +1,97 @@
+/* This file is part of Indico.
+ * Copyright (C) 2002 - 2015 European Organization for Nuclear Research (CERN).
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+(function($) {
+    $.widget('indico.colorswitch', {
+
+        options: {
+            defaultColors: [],
+            onSelect: null,
+            numColumns: 5
+        },
+
+        _create: function() {
+            var self = this;
+            var element = this.element;
+            var palette = $('<div>', {'class': 'color-palette', 'css': {'display': 'none'}});
+            var paletteTable = $('<table>');
+            var trigger = this.element.find('.switch-trigger');
+            var defaultColors = this.options.defaultColors;
+            var tr = this._createTableRow();
+
+            for (var i = 0; i < defaultColors.length; ++i) {
+                var td = $('<td>', {
+                    'css': {'background': '#' + defaultColors[i].background},
+                    'class': 'palette-color'
+                });
+
+                var colorBox = $('<div>', {
+                    'value': '#' + defaultColors[i].background,
+                    'class': 'background-box'
+                });
+
+                colorBox.append($('<div>', {
+                    'css': {'background': '#' + defaultColors[i].text},
+                    'class': 'text-box',
+                    'value': '#' + defaultColors[i].text
+                }));
+
+                td.on('click', function() {
+                    var $this = $(this),
+                        backgroundColor = $this.find('.background-box').val(),
+                        textColor = $this.find('.text-box').val();
+
+                    trigger.css({'background': backgroundColor, 'color': textColor + ' !important'});
```

Comment:
no need to quote the keys (only `class` requires it for certain sub-par "browsers", but in dicts where there's no `'class'` key there's no need to quote any keys)